### ISSUE TITLE
cannot uninstall

### ISSUE THREAD

**Issue by riley8084** [jlobue10/SteamDeck_rEFInd#30]
having this issue where im unable to uninstall. (see attached screen shot)

![ksnip_20230118-140427](https://user-images.githubusercontent.com/123016427/213271399-04439571-2caa-4edf-89ad-6164e4009376.png)

**Comment by riley8084** [comment]
also if i  go to install it again it says its already installed so im wondering why it cant find it.

**Comment by jlobue10** [comment]
I think I may see the issue. Can you please try these one by one?

```
sudo steamos-readonly disable
sudo pacman -Syu
sudo pacman -R refind
sudo steamos-readonly enable
```
My best guess (from what you've shown me) is that you are trying to run the package removal command at the same time you are revoking root write permissions. This will not work as those permissions are needed for pacman package changes. You need to do the removal step, let it finish and then re-enable the write protections. Basically do what I put above, one line at a time and let me know if it works or not. Also, out of curiosity, why are you uninstalling?

**Comment by riley8084** [comment]
same issue again unfortunately.

reason im uninstalling is because when i first installed it worked fine but only for the first boot.
every other boot after that it booted into the windows OS.
i tried a fix on youtube where i changed my steam deck to accept experimental updates or whatever and that fixed it for like another 3-4 boots
now its back to booting to just windows.
i can see it in the boot menu when i select volume down and power.
the boot order is probably messed up or something but im not really that advanced enough to change it.
cant even uninstall it now either lol.

![ksnip_20230118-182432](https://user-images.githubusercontent.com/123016427/213317304-84e38459-76be-4c69-b1b3-db0f008dfcaf.png)

**Comment by jlobue10** [comment]
That is really strange and not the behavior that I'd expect from `pacman`. Hmm... Maybe I can help you fix your installation, if you're interested in that. I have made significant script improvements within the last few weeks. If you'd be interested in trying to fix your installation, before continuing with attempts to uninstall, please delete your cloned directory, re-clone the repository and go through the README steps for reinstallation. The reason that Windows took over after the last update was most likely due to the BIOS update that came alongside of it. Please let me know if you'd like to try to fix the installation, or if your mind is completely made up on continuing to try other things to uninstall.

I've also made good progress on the GUI, and I hope to release it soon for more control over customization.

**Comment by riley8084** [comment]
I'd definitely be down to give it shot! 
Whatever it takes to make it work :) 

Where can I find those directories? 
I apologize in advance, I'm not the MOST versed with Linux and all of that.

**Comment by jlobue10** [comment]
Yes, no worries. I aim to support people of all experience levels, since having a functional rEFInd bootloader is just a very nice QoL improvement for dual boot on the Steam Deck (in my opinion).

The README.md file should load automatically for you if you just open the main [GitHub repo page](https://github.com/jlobue10/SteamDeck_rEFInd). I actually just updated the no-pacman installation script too, since I forgot to update it this morning to coincide with a change I made to the `pacman` version. Please let me know if you have any questions. I try to be as thorough as possible, without being too loquacious in the README. :)

**Comment by riley8084** [comment]
Really appreciate your responsiveness, means a lot!

I just did what you recommended, deleted the folder, recloned and then git status. 
Worked for when I chose steam but then whenever I select windows it only boots to windows every time after that :/ 

Could it be that it's my windows version? I was watching some vids where people said you could just get the windows set up tool off the website.

Just tryna think of reasons why I'm getting weird little errors

**Comment by jlobue10** [comment]
No, this is progress, and exactly what I expect! What is almost certainly happening, is that your Windows EFI entry is active. This can be ignored and left active if you successfully set up the Windows scheduled task with the provided powershell script, or you can disable the Windows EFI entry using EasyUEFI trial from Windows (or both). Hopefully the README is clear about both of those things. If you boot into SteamOS desktop mode and run `efibootmgr` from the command line, this would likely confirm my suspicion about the Windows EFI entry being active, just as an fyi.

**Comment by riley8084** [comment]
here is a screen shot

most of this is foreign to me lol.

![ksnip_20230118-195417](https://user-images.githubusercontent.com/123016427/213330521-c1b2671a-f828-449a-926d-fce1496ede1e.png)

**Comment by jlobue10** [comment]
Yes, so this screen actually tells me a lot. The BootNext 0000 (rEFInd's boot number) tells me that the systemd service is doing it's job properly. I also see both SteamOS and rEFInd EFI entries are present and correct. The part that confirms my suspicion about the Windows EFI entry being active is that asterisk next to Boot0002*. The Windows EFI entry needs to either be inactive (2 methods of accomplishing this are talked about in the README), or you need to perform the one time manual setup of the PowerShell scheduled task in Windows (or you can do both those things!). If the Windows EFI entry is active and you don't have the scheduled task setup in Windows, then Windows will assume the top boot priority. The firmware just really wants to boot into Windows, if this makes sense. It's actually been somewhat of a pain, but it's mostly managed now, since it is a known issue.

**Comment by riley8084** [comment]
i see you mentioned disabling windows in the steam recovery boot using the command line, i didn't see anything regarding how to do so, just how to do it through windows. what is easiest in your opinion?

**Comment by jlobue10** [comment]
Honestly, both methods are easy. If you have the Steam recovery USB available and know how to force boot from it (from BIOS), then the command you would run for your particular EFI entry table is `sudo efibootmgr -b 0002 -A` . If you do this properly, you will see the asterisk go away next to Boot0002. Using EasyUEFI trial from Windows is also easy and is done with a GUI. You select the Windows entry and then just click the disable icon.

**Comment by riley8084** [comment]
![image](https://user-images.githubusercontent.com/123016427/213336432-c99d5053-141e-4773-9e0d-1879faa6581d.jpeg)

Does this look right?

**Comment by jlobue10** [comment]
It needs to be a capital -A, like `sudo efibootmgr -b 0002 -A` and then you should be good (and asterisk will be gone).

**Comment by riley8084** [comment]
Okay it worked that time! 

I can confirm after booting into windows that it still goes back to reFInd after

Now to bug you with TWO more questions 

First being, I still have this issue where it only stays on the boot menu for a couple of seconds and then defaults to steam. Is there a way to make it where it doesn't time out?

And secondly how do I change the background for the boot screen? I saw there were a few jpegs in the files there, wanted to try out the water one.

Again thank you so much for your responsiveness. 

I can't recommend you enough after how you've helped me figure this out!

**Comment by riley8084** [comment]
Okay so actually I narrowed down the issue a bit more.

When I boot the deck up and don't touch the cursor, the boot menu will go away after roughly 3 seconds

When I boot up and I live the cursor around it doesn't go away.

So with no input it just defaults to the steam OS

**Comment by jlobue10** [comment]
When you say, "Is there a way to make it where it doesn't time out?" Do you mean that you want to select the OS that boots every time manually? If this is what you mean, then set `timeout 0` in the `refind.conf` config file. I have the default config file set for a timeout of 5 seconds and the default OS as SteamOS. If no user input is detected during those 5 seconds, then SteamOS will boot. I figure this is close to what a lot of people would want.

To change to the water picture background, you will need to make another edit to the `refind.conf` config file. The portion that looks like this

```
#banner hostname.bmp
#banner mybanner.jpg
#banner icons/snowy/banner-snowy.png
banner themes/background.png
#banner themes/background2.png
#banner themes/background3.png
#banner themes/background4.png
#banner themes/background5.png
#banner themes/background6.png
#banner themes/background7.png
```
will need to be changed to look like this.
```
#banner hostname.bmp
#banner mybanner.jpg
#banner icons/snowy/banner-snowy.png
#banner themes/background.png
#banner themes/background2.png
#banner themes/background3.png
#banner themes/background4.png
banner themes/background5.png
#banner themes/background6.png
#banner themes/background7.png
```
Basically, we are commenting out the active background line (adding a `#` at the beginning of the line) and making the background5.png line active (deleting the `#`).

You have 2 easy options to edit the config file. Either edit the config file in the cloned repo folder and then re-run the installation script, or from a command line run `sudo nano /esp/efi/refind/refind.conf`, make the edits and then press `Ctrl+x` followed by `y` then `Enter`.

I'm happy to help. Thank you for having the patience for us to fix your issues and keep the installation (after it was fixed).

**Comment by riley8084** [comment]
Really blown away by your level of expertise and attention to detail. 
Beyond happy that I chose to keep this boot loader and not uninstall. 
Also I will be recommending your script to anyone who's looking for a boot loader in the future!

Thank you again for taking the time to help me. You have made my experience so much more enjoyable!

**Comment by jlobue10** [comment]
Yes, no worries. This is why I asked, since most people who have it setup properly do not want to uninstall! I'm happy that we got it working for you. Another satisfied customer of my free script. XD. But in all seriousness, I want people to be happy with the script. This stuff is easy for me. I realize for other people, it may not be easy and may be frustrating, which is why I do my best to support it. Once the GUI is ready, customization will be even easier for everyone (setting backgrounds, OS icons, timeout, etc.). The going (with the GUI development) is a bit slow though since I'm not a professional Qt developer and it's been since college that I really coded much in C++. With that being said though, I have made some good progress on the GUI in just the last few days.

If you're happy with the resolution, please close the issue.

**Comment by riley8084** [comment]
thankyou for all that you do!

very happy with the resolution, I will go ahead and close the issue!
